test/nn/test_pooling.py

```diff
@@ -153,6 +153,23 @@ def test_adaptive_pooling_size_none(self):
                 output = module(input)
                 self.assertEqual(output.size(), (4,) + (2,) * (numel - 1) + (4,))
 
+    def test_adaptive_pooling_empty_output_size(self):
```

Comment:
For test on multiple device, put this test under `TestPoolingNNDeviceType`.

Reply:
On second thought, I think it would be better to move these tests to OpInfo `error_inputs`. That way other sub-systems can also test for this fix.

Reply:
As per offline discussion with @kshitij12345, the ErrorInputs cannot be used in cases where forward mode works well and backward raises an error. Hence, shifted the test under `TestPoolingNNDeviceType`. Thanks :)